### PR TITLE
User component

### PR DESCRIPTION
**Pull request by EVDW**

### CODE REVIEW

File: _includes/_components/user.html [hunk 1]
```diff
@@ -4,13 +4,10 @@
         <header>
           <div class="container">
             <h2>Users</h2>
-            <p>This component blablabla</p>
+            <p>A simply way to handle password encryption and salt generation.</p>
```
Comment (einenlum) [inline _includes/_components/user.html]: simple way

File: _includes/_components/user.html [hunk 2]
```diff
@@ -1,51 +1,150 @@
...
+    // ...
+
+    public function setPassword($password)
+    {
+        $this->password = $password;
+
+        return $this->password;
+    }
+
+    public function setSalt($salt)
```
Comment (einenlum) [inline _includes/_components/user.html]: See [here](http://symfony.com/doc/current/cookbook/security/entity_provider.html).

You generally don't add a setSalt() method. You either generate a unique salt in constructor or you just don't (if you use bcrypt).

File: _includes/_components/user.html [hunk 3]
```diff
@@ -1,51 +1,150 @@
...
+
+    private $salt;
+
+    // ...
+
+    public function setPassword($password)
+    {
+        $this->password = $password;
+
+        return $this->password;
```
Comment (einenlum) [inline _includes/_components/user.html]: `return $this;`

File: _includes/_components/user.html [hunk 4]
```diff
@@ -1,51 +1,150 @@
...
+
+use App\Controller\Controller;
+use App\Entity\User;
+
+class UserController extends Controller
+{
+    public function newUserAction($email,$plainPassword)
+    {
+        $user = new User();
+        $salt = uniqid(mt_rand());
```
Comment (einenlum) [inline _includes/_components/user.html]: see my comment above

File: _includes/_components/user.html (hunk 3 above)
Comment (EVDW) [inline _includes/_components/user.html]: Oups :D

**Comment by Einenlum** [comment]
Please squash.

**Pull request merged**